cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/PluginInstantiator.java

```diff
@@ -268,9 +269,17 @@ private Object convertValue(TypeToken<?> fieldType, String value) throws Excepti
         rawType = Primitives.wrap(rawType);
       }
 
-      if (Primitives.isWrapperType(rawType)) {
-        Method valueOf = rawType.getMethod("valueOf", String.class);
-        return valueOf.invoke(null, value);
+      try {
```

Comment:
I would prefer the `try-catch` block be more precise, meaning moving it inside the `if` block, just around the `invoke` call.